modules/exadsBidAdapter.js

```diff
@@ -0,0 +1,433 @@
+import * as utils from '../src/utils.js';
+import { BANNER, NATIVE, VIDEO } from '../src/mediaTypes.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+
+const BIDDER = 'exadsadserver';
+
+const PARTNERS = {
+  RTB_2_4: 'rtb_2_4'
+};
+
+const envParams = {
+  lang: '',
+  userAgent: '',
+  osName: '',
+  page: '',
+  domain: '',
+  language: '',
+  userId: ''
+};
+
+const adPartnerHandlers = {
+  [PARTNERS.RTB_2_4]: {
+    request: handleReqRTB2Dot4,
+    response: handleResRTB2Dot4,
+    validation: handleValidRTB2Dot4,
+  }
+};
+
+function handleReqRTB2Dot4(bid, endpointUrl, validBidRequests, bidderRequest) {
```

Comment:
I don't think using `bid` is correct here. I understood that we should only consider `validBidRequests`

Reply:
Actually "bid" is a valid bid. To see the functions calling handleReqRTB2Dot4.

On the other side validBidRequests is the array of all valid request (potentially you could manage multiple requests).

Considering that I'm not using "validBidRequests" I've renamed bid to validBidRequest and removed validBidRequests.

Now we'll have:

```suggestion
function handleReqRTB2Dot4(validBidRequest, endpointUrl, bidderRequest) {
```

Now it should be more clear.